akka-projection-slick/src/main/scala/akka/projection/slick/OffsetStore.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.slick
+
+import java.util.UUID
+
+import akka.Done
+import akka.annotation.ApiMayChange
+import slick.jdbc.JdbcProfile
+import akka.persistence.query
+
+import scala.concurrent.Future
+
+@ApiMayChange
+class OffsetStore[P <: JdbcProfile](db: P#Backend#Database, profile: P) {
+
+  import profile.api._
+
+  // offset manifest
+  private val StringM = "STR"
+  private val LongM = "LNG"
+  private val IntM = "INT"
+  private val SequenceM = "SEQ"
+  private val TimeBasedUUIDM = "TBU"
```

Comment:
Those are the supported types of offsets. This is light version of manifest. We use to keep the type hint and be able to read it back.

Reply:
Correct. We can add it later though. We may also consider moving it to a utility class that can be resued by Cassandra offset store. 

For `java.sql.Timestamp` we will need to have a concatenated string `long:long`. Hacky, but that will simplify our lives.